src/init.cpp

```diff
@@ -1361,11 +1368,14 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
     if (onion_proxy.IsValid()) {
         SetProxy(NET_ONION, onion_proxy);
     } else {
-        if (args.IsArgSet("-onlynet") && IsReachable(NET_ONION)) {
+        // If -listenonion is set, then we will (try to) connect to the Tor control port
+        // later from the torcontrol thread and may retrieve the onion proxy from there.
+        const bool maybe_later = args.GetBoolArg("-listenonion", DEFAULT_LISTEN_ONION);
+        if (onlynet_used_with_onion && !maybe_later) {
             return InitError(
                 _("Outbound connections restricted to Tor (-onlynet=onion) but the proxy for "
-                  "reaching the Tor network is not provided (no -proxy= and no -onion= given) or "
-                  "it is explicitly forbidden (-onion=0)"));
+                  "reaching the Tor network is not provided: none of -proxy, -onion or "
+                  "-listenonion is given"));
```

Comment:
Maybe consider something like this as a more helpful error message:
```diff
-                _("Outbound connections restricted to Tor (-onlynet=onion) but the proxy for "
-                  "reaching the Tor network is not provided: none of -proxy, -onion or "
-                  "-listenonion is given"));
+                _("The -onlynet=onion configuration option was passed to restrict outbound"
+                  " connections to Tor, but the proxy for reaching the Tor network was not"
+                  " provided: none of -proxy, -onion or -listenonion was given"));
```

Reply:
Done.